File: testapp/docs/main_fact.cpp
#include <stdio.h>
#include <stdlib.h>

extern int factorial(int);

template <class T>

void check(T expect, T result)
{
    if (expect == result)
    {
	//printf("Correct:\n Expected %d got %d \n",expect,result);
    }
    else 
    {
	printf("Incorrect:\n Expected %d got %d \n",expect,result);
	exit (0);
   }
}

int main(void)
{
	int result;
	result = factorial(0);
	check(1, result);
	result = factorial(3);
	check(6, result);
	printf("All Correct\n");
}

File: testapp/java_files/main_fact.java
class main_fact
{
    public static <E> void check(E expect, E result)
    {
        if(result.equals(expect))
        {
            System.out.println("Correct:\nOutput expected "+expect+" and got "+result);
        }
	else
	{
	    System.out.println("Incorrect:\nOutput expected "+expect+" but got "+result);
	    System.exit(1);
	}
    }
    public static void main(String arg[])
    {
       Test t = new Test();
       int result;
       result = t.factorial(0);
       System.out.println("Input submitted to the function: 0");
       check(1, result);
       result = t.factorial(3);
       System.out.println("Input submitted to the function: 3");
       check(6, result);
       result = t.factorial(4);
       System.out.println("Input submitted to the function: 4");
       check(24, result);
    }
}
